src/core/SpriteAnimator.tsx

```diff
@@ -1,72 +1,176 @@
   endFrame?: number
+  /** The desired frames per second of the animation. If set to 0 or negative, animation will be static */
   fps?: number
+  /** The frame identifier to use, must be one of animationNames */
   frameName?: string
+  /** The URL of the texture JSON (if using JSON-Array or JSON-Hash) */
   textureDataURL?: string
+  /** The URL of the texture image */
   textureImageURL?: string
+  /** Whether or not the animation should loop */
   loop?: boolean
+  /** The number of frames of the animation (required if using plain spritesheet without JSON) */
   numberOfFrames?: number
+  /** Whether or not the animation should auto-start when all assets are loaded */
   autoPlay?: boolean
+  /** The animation names of the spritesheet (if the spritesheet -with JSON- contains more animation sequences) */
   animationNames?: Array<string>
-  onStart?: Function
-  onEnd?: Function
-  onLoopEnd?: Function
-  onFrame?: Function
+  /** Event callback when the animation starts or restarts
+   * @param data Animation state data containing currentFrameName and currentFrame
+   */
+  onStart?: (data: AnimationEventData) => void
+  /** Event callback when the animation ends
+   * @param data Animation state data containing currentFrameName and currentFrame
+   */
+  onEnd?: (data: AnimationEventData) => void
+  /** Event callback when the animation completes a loop cycle
+   * @param data Animation state data containing currentFrameName and currentFrame
+   */
+  onLoopEnd?: (data: AnimationEventData) => void
+  /** Event callback fired on each frame change
+   * @param data Animation state data containing currentFrameName and currentFrame
+   */
+  onFrame?: (data: AnimationEventData) => void
+  /** @deprecated Use pause={false} instead. Control when the animation runs */
   play?: boolean
+  /** Control when the animation pauses */
   pause?: boolean
+  /** Whether or not the Sprite should flip sides on the x-axis */
   flipX?: boolean
-  position?: Array<number>
+  /** Sets the alpha value to be used when running an alpha test
+   * @see https://threejs.org/docs/#api/en/materials/Material.alphaTest
+   * @default 0.0
+   */
   alphaTest?: number
+  /** Displays the texture on a Billboard component always facing the camera.
+   * If set to false, it renders on a PlaneGeometry
+   * @default true
+   */
   asSprite?: boolean
+  /** Allows for manual update of the sprite animation e.g: via ScrollControls.
+   * Value should be between 0 and 1
+   */
   offset?: number
+  /** Allows the sprite animation to start from the end towards the start */
   playBackwards?: boolean
+  /** Allows the animation to be paused after it ended so it can be restarted on demand via autoPlay */
   resetOnEnd?: boolean
+  /** Array of Vector3-like positions for creating multiple instances of the sprite */
+  instanceItems?: (THREE.Vector3 | [number, number, number])[]
+  /** The maximum number of instances to render (for buffer size calculation)
+   * @default 1
```

Comment:
to avoid this `@default` manual jsdoc annotation, you could rely on TS which will infer it as long as you really use default value in the props:

```ts
const SpriteAnimator = ...
(
    {
      startFrame,
      ...
      maxItems = 1, // default value !
      ...
    },
    fref
  ) => {
```

if using a real default value like this, Storybook will have it also in its doc

Reply:
<img width="1800" alt="image" src="https://github.com/user-attachments/assets/c0f709a5-474b-4a1d-be71-1c039930aec9">

Reply:
+ same for other props with default values

Reply:
done in latest commit